MapleServer2/Types/SkillCast.cs

```diff
@@ -42,22 +46,66 @@ public SkillCast(int id, short level, long skillSN, int unkValue)
             UnkValue = unkValue;
         }
 
-        public double GetDamageRate() => SkillMetadataStorage.GetSkill(SkillId).SkillLevels.Find(x => x.Level == SkillLevel).DamageRate;
+        public double GetDamageRate() => SkillMetadataStorage.GetSkill(SkillId)?.SkillLevels.Find(x => x.Level == SkillLevel).DamageRate ?? 0.1f;
 
         public double GetCriticalDamage() => 2 * GetDamageRate();
 
-        public int GetSpCost() => SkillMetadataStorage.GetSkill(SkillId).SkillLevels.Find(s => s.Level == SkillLevel).Spirit;
+        public int GetSpCost() => SkillMetadataStorage.GetSkill(SkillId)?.SkillLevels.Find(s => s.Level == SkillLevel).Spirit ?? 15;
 
-        public int GetStaCost() => SkillMetadataStorage.GetSkill(SkillId).SkillLevels.Find(s => s.Level == SkillLevel).Stamina;
+        public int GetStaCost() => SkillMetadataStorage.GetSkill(SkillId)?.SkillLevels.Find(s => s.Level == SkillLevel).Stamina ?? 10;
 
-        public DamageTypeId GetSkillDamageType() => (DamageTypeId) SkillMetadataStorage.GetSkill(SkillId).DamageType;
+        public DamageTypeId GetSkillDamageType()
+        {
+            if (SkillMetadataStorage.GetSkill(SkillId) != null)
+            {
+                return (DamageTypeId) SkillMetadataStorage.GetSkill(SkillId).DamageType;
+            }
+            else
+            {
+                return DamageTypeId.None;
+            }
```

Comment:
Maybe it's too dank...

```suggestion
            return SkillMetadataStorage.GetSkill(SkillId) != null ? (DamageTypeId) SkillMetadataStorage.GetSkill(SkillId).DamageType : DamageTypeId.None;
```

Reply:
Even more dank

`        public DamageTypeId GetSkillDamageType() => SkillMetadataStorage.GetSkill(SkillId) != null ? (DamageTypeId) SkillMetadataStorage.GetSkill(SkillId).DamageType : DamageTypeId.None;
`

Reply:
i recently add `GetSkillMetadata()` to reduce the calls and more readable, suggest the change with it so i can commit it